pkg/controller/grouppermission/grouppermission_controller.go

```
@@ -100,54 +86,196 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
-		return reconcile.Result{}, err
+	// get list of clusterRole on k8s
+	clusterRoleList := &v1.ClusterRoleList{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleList)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, nil
 	}
 
-	// Check if this Pod already exists
-	found := &corev1.Pod{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
-		err = r.client.Create(context.TODO(), pod)
+	// checkClusterRoleName is a helper func, returns bool
+	if checkClusterRoleName(instance, clusterRoleList) {
+		return reconcile.Result{}, nil
+	} else {
+		// get the conditions array
+		groupPermissionConditions := instance.Status.Conditions
+
+		// make new condition
+		newCondition := managedv1alpha1.Condition{
+			LastTransitionTime: metav1.Now(),
+			Message:            "ClusterRole for clusterPermission does not exist",
+			ClusterRoleName:    "clusterRoleName", //this needs to get fixed
+			Status:             true,
+			State:              "Failed",
+		}
+
+		// append new condition to conditions array
+		groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+		// update GroupPermission object with new conditions
+		instance.Status.Conditions = groupPermissionConditions
+
+		err = r.client.Status().Update(context.TODO(), instance)
 		if err != nil {
-			return reconcile.Result{}, err
+			reqLogger.Error(err, "Failed to update condition.")
+			return reconcile.Result{}, nil
 		}
+	}
 
-		// Pod created successfully - don't requeue
+	clusterRoleBindingList := &v1.ClusterRoleBindingList{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleBindingList)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Delete this clause and body.  If the err exists it should return an error.